### PR TITLE
OpticalFlow: Added Pixart 3900 optical flow sensor driver

### PR DESCRIPTION
**Pull request by tridge**
This is a work in progress driver for the Pixart PMW3900 SPI optical flow sensor.
This needs some re-work to cleanup the commits. The driver is working though, and I've had good test flights.

### CODE REVIEW

File: libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+#include <AP_Math/edc.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <utility>
+#include "OpticalFlow.h"
+#include "AP_OpticalFlow_Pixart_SROM.h"
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define PIXART_REG_PRODUCT_ID  0x00
```
Comment (muramura) [inline libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp]: Register numbers are grouped and I think the other is good.

**Comment by tridge** [comment]
as a bonus this now adds an in-tree PX4Flow driver common between linux and PX4 boards

**Comment by tridge** [comment]
I think this PR is ready for master now, pending review

**Comment by OXINARF** [comment]
I just checked the PX4Flow and I saw you removed the padding added by https://github.com/ArduPilot/ardupilot/pull/4810. Have you confirmed this works? Please read the PR as me and Lucas reviewed it and it looked like the padding was the correct thing to do (or changing PX4Flow firmware but that doesn't look like an available option).

**Comment by OXINARF** [comment]
Also, this is includes PR #5241, has that been cleared to go in? From all the comments, including Leonard's I would say it needs to be changed.

**Comment by tridge** [comment]
@OXINARF you are right about #5241 I'll fix that up
regarding the padding, that is not needed. You can ready 1 byte less in an i2c read with no issues. I have tested this on a px4flow and its fine.

**Comment by tridge** [comment]
I've removed the copter glitch code and applied

**Pull request merged**